templates/myid.j2

```diff
@@ -1,10 +1,5 @@
-{%- if zookeeper.id is defined -%}
-{{ zookeeper.id  }}
+{%- if zookeeper.aws_cluster_autodiscover.data is defined -%}
+{{ zookeeper.aws_cluster_autodiscover.data.id }}
 {%- else -%}
-  {%- for url in zookeeper.hosts -%}
-    {%- set url_host = url.split(':')[0] -%}
-    {%- if url_host == ansible_fqdn or url_host in ansible_all_ipv4_addresses -%}
-{{ loop.index0  }}
-    {%- endif -%}
-  {%- endfor -%}
+{{ zookeeper.id }}
 {%- endif -%}
```

Comment:
Similar to the kafka PR, I need to detect that that parent ookeeper.aws_cluster_autodiscover.data element exists otherwise local VMs will complain.